src/main/scala/io/gatling/sbt/settings/gatling/EnterpriseSettings.scala

```diff
@@ -0,0 +1,175 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gatling.sbt.settings.gatling
+
+import java.io.File
+import java.util.UUID
+
+import scala.collection.JavaConverters._
+
+import io.gatling.plugin.util.OkHttpEnterpriseClient
+import io.gatling.sbt.BuildInfo
+import io.gatling.sbt.GatlingKeys._
+import io.gatling.sbt.utils.{ DependenciesAnalysisResult, DependenciesAnalyzer, FatJar }
+
+import sbt._
+import sbt.Keys._
+
+object EnterpriseSettings {
+  private def moduleDescriptorConfig = Def.task {
+    moduleSettings.value match {
+      case config: ModuleDescriptorConfiguration => config
+      case x =>
+        throw new IllegalStateException(s"gatling-sbt expected a ModuleDescriptorConfiguration, but got a ${x.getClass}")
+    }
+  }
+
+  private def buildEnterprisePackage(config: Configuration): Def.Initialize[Task[File]] = Def.task {
+    val moduleDescriptor = moduleDescriptorConfig.value
+
+    val DependenciesAnalysisResult(gatlingVersion, dependencies) = DependenciesAnalyzer.analyze(
+      dependencyResolution.value,
+      updateConfiguration.value,
+      (update / unresolvedWarningConfiguration).value,
+      config,
+      streams.value.log,
+      moduleDescriptor
+    )
+
+    val classesDirectories = (config / fullClasspath).value.map(_.data).filter(_.isDirectory)
+
+    val jarName = s"${moduleName.value}-gatling-enterprise-${version.value}"
+
+    FatJar
+      .packageFatJar(moduleDescriptor.module, classesDirectories, gatlingVersion, dependencies, target.value, jarName)
+  }
+
+  private def legacyPackageEnterpriseJar(config: Configuration): Def.Initialize[Task[File]] = Def.sequential(
+    Def.task {
+      val newCommand = config.id match {
+        case Test.id            => "Gatling / enterprisePackage"
+        case IntegrationTest.id => "GatlingIt / enterprisePackage"
+        case _                  => "Gatling / enterprisePackage or GatlingIt / enterprisePackage"
+      }
+      streams.value.log.warn(
+        s"""Task ${config.id} / assembly is deprecated and will be removed in a future version.
+           |Please use $newCommand instead.
+           |See https://gatling.io/docs/gatling/reference/current/extensions/sbt_plugin/ for more information.""".stripMargin
+      )
+    },
+    buildEnterprisePackage(config)
+  )
+
+  private val PublicApiPath = "/api/public"
+
+  private def httpEnterpriseClient(config: Configuration) = Def.task {
+    val settingUrl = new URL((config / enterpriseUrl).value.toExternalForm + PublicApiPath)
+    val settingApiToken = (config / enterpriseApiToken).value
+
+    if (settingApiToken.isEmpty) {
+      throw new IllegalStateException("Gatling / apiToken has not been specified")
+    }
+
+    val client = new OkHttpEnterpriseClient(settingUrl, settingApiToken)
+    client.checkVersionSupport(BuildInfo.version, BuildInfo.name)
```

Comment:
it should be checkVersionSupport(name, version)

Reply:
named parameters?